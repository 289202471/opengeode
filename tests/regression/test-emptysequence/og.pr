SYSTEM og;
    /* CIF TEXT (159, 221), (289, 188) */
    -- Text area for declarations and comments

use dv comment 'dataview.asn';

signal dd;
signal we;
    /* CIF ENDTEXT */
    CHANNEL c
        FROM ENV TO og WITH dd;
        FROM og TO ENV WITH we;
    ENDCHANNEL;
    BLOCK og;
        SIGNALROUTE r
            FROM ENV TO og WITH dd;
            FROM og TO ENV WITH we;
        CONNECT c AND r;
        /* CIF PROCESS (225, 49), (150, 75) */
        PROCESS og;
            /* CIF TEXT (57, 58), (337, 130) */
            --  Test support for empty sequences and NULL type

dcl empty1 EmptySeq := c1;

dcl null1 NullType := NULL;

dcl choice1 ChoiceWithEmptySeq := a: n1;
            /* CIF ENDTEXT */
            /* CIF START (160, 226), (70, 35) */
            START;
                /* CIF TASK (137, 276), (115, 35) */
                TASK empty1 := c1
                /* CIF COMMENT (272, 276), (139, 35) */
                COMMENT 'Using a constant';
                /* CIF TASK (138, 326), (112, 35) */
                TASK empty1 := {}
                /* CIF COMMENT (270, 326), (135, 35) */
                COMMENT 'Classic notation';
                /* CIF TASK (130, 376), (130, 35) */
                TASK choice1 := b: c1;
                /* CIF TASK (131, 426), (126, 35) */
                TASK choice1 := b: {};
                /* CIF TASK (144, 476), (100, 35) */
                TASK null1 := n1;
                /* CIF TASK (134, 526), (121, 35) */
                TASK null1 :=  null;
                /* CIF NEXTSTATE (160, 576), (70, 35) */
                NEXTSTATE wait;
            /* CIF STATE (424, 231), (70, 35) */
            STATE wait;
                /* CIF INPUT (417, 286), (84, 35) */
                INPUT dd;
                    /* CIF OUTPUT (414, 336), (89, 35) */
                    OUTPUT we;
                    /* CIF NEXTSTATE (424, 386), (70, 35) */
                    NEXTSTATE wait;
            ENDSTATE;
        ENDPROCESS og;
    ENDBLOCK;
ENDSYSTEM;