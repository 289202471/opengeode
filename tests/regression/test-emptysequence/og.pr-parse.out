[INFO] Starting to parse model: test-emptysequence/og.pr
[ERROR] Failed to parse model.
    line 3:4 no viable alternative at input '/* CIF TEXT (159, 221), (289, 188) */\n    -- Text area for declarations and comments\n\nuse dv comment 'dataview.asn';\n\nsignal dd;\nsignal we;\n    /* CIF ENDTEXT */\n    CHANNEL c\n        FROM ENV TO og WITH dd;\n        FROM og TO ENV WITH we;\n    ENDCHANNEL;\n    BLOCK og;\n        SIGNALROUTE r\n            FROM ENV TO og WITH dd;\n            FROM og TO ENV WITH we;\n        CONNECT c AND r;\n        /* CIF PROCESS (225, 49), (150, 75) */\n        PROCESS og;\n            /* CIF TEXT (79, 58), (337, 130) */\n            --  Test support for empty sequences and NULL type\n\ndcl empty1 EmptySeq := c1;\n\ndcl null1 NullType;\n\ndcl choice1 ChoiceWithEmptySeq := a: n1;\n            /* CIF ENDTEXT */\n            /* CIF START (182, 226), (70, 35) */\n            START;\n                /* CIF TASK (160, 276), (115, 35) */\n                TASK empty1 := c1\n                /* CIF COMMENT (294, 276), (139, 35) */\n                COMMENT 'Using a constant';\n                /* CIF TASK (161, 326), (112, 35) */\n                TASK empty1 := {}\n                /* CIF COMMENT (292, 326), (135, 35) */\n                COMMENT 'Classic notation';\n                /* CIF TASK (152, 376), (130, 35) */\n                TASK choice1 := b: c1;\n                /* CIF TASK (154, 426), (126, 35) */\n                TASK choice1 := b: {};\n                /* CIF TASK (167, 476), (100, 35) */\n                TASK null1 := n1;\n                /* CIF NEXTSTATE (182, 526), (70, 35) */\n                NEXTSTATE wait;\n            /* CIF STATE (446, 231), (70, 35) */\n            STATE wait;\n                /* CIF INPUT (439, 286), (84, 35) */\n                INPUT dd;\n                    /* CIF OUTPUT (437, 336), (89, 35) */\n                    OUTPUT we;\n                    /* CIF NEXTSTATE (446, 386), (70, 35) */\n                    NEXTSTATE wait;\n            ENDSTATE;\n        ENDPROCESS og;\n    ENDBLOCK;\n'
